w1/assignment2_3/main.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <cmath>
+
+struct StudentStruct
+{
+    std::string name {};
+    int age {};
+    int grade {};
+};
+
+void set_students(std::vector<StudentStruct> &students_vect)
+{
+    StudentStruct student;
+    std::cout << "Count of students: " << std::endl;
+    int count {};
+    std::cin >> count;
+    for(int i = 0; i < count; i++){
+        std::cout << "Name of student: " << std::endl;
+        std::cin >> student.name;
+        std::cout << "Age of student: " << std::endl;
+        std::cin >> student.age;
+        std::cout << "Grade of student: " << std::endl;
+        std::cin >> student.grade;
+        students_vect.push_back(student);
+    }
+}
+
+void print_average(std::vector<StudentStruct> &students_vect)
+{
+    double sum {0};
+    for(StudentStruct student : students_vect)
```

Comment:
Here you could make use of pass-by-reference, to avoid unecessary copy of students_vect:

for (StudentStruct& student : students_vect)
or for(auto& student : ...